Add a tutorial on porting test groups from legacy

xylar opened the pull request:
This tutorial uses the `gotm` test group and the default test case as an example.

xylar commented:
@mark-petersen and @matthewhoffman, in the spirit of being ready for new students, I made the first of what I expect to be several tutorials in the compass documentation.  This one is on porting a legacy test group.  Given that it took me 3 days (off an on) to put together, these tutorials aren't going to come all that quickly.  But maybe this is one of the more critical?

Anyway, I appreciate any feedback.  The easiest way is if you actually edit the rst file using the "suggest" feature in GitHub.   Highlight the relevant lines and, in your comment, hit the suggest button then edit the text. That way, I can aggregate suggested changes here and make a commit.

xylar commented:
@qingli411, it would also be helpful to have your review on this one, and reading this tutorial might give you a better handle on reviewing #129

matthewhoffman commented:
@xylar , can you remind me how to build the documentation in this branch for easier reading?

xylar commented:
Sure:
```
source test_compass_1.0.0*.sh
cd docs
make html
google-chrome _build/html/index.html
```

xylar commented:
@vanroekel ran into trouble with the `conda/configure_compass_env.sh` in the SCORPIO build on his Mac (and I saw it, too).  I'm working on that as I have time but in the meantime you might need to copy the `_build/html` directory over to somewhere that you can open it in a browser.

mark-petersen reviewed (approved):
@xylar I looked over the rst files. It looks good. Thanks for adding this documentation.

qingli411 commented on docs/tutorials/dev_porting_legacy.rst:
```diff
@@ -0,0 +1,1296 @@
+
+.. note::
+
+    In this particular test case, the there are no land cells defined and
```
```suggestion
    In this particular test case, there are no land cells defined and
```

qingli411 commented on docs/tutorials/dev_porting_legacy.rst:
```diff
@@ -0,0 +1,1296 @@
+Make sure to add some documentation of your new test group.  You need to add
+all of the functions, classes and methods to the API documentation in
+``docs/developers_guide/<core>/api.rst``, following the exmaples for other
+test groups.  You also need to add a file to both the user's guid and the
```
```suggestion
test groups.  You also need to add a file to both the user's guide and the
```

qingli411 reviewed (approved):
@xylar This tutorial is really helpful! Thanks for creating it! It reads very well and is very informative. I only find a few typos which are highlighted in the suggested changes.

xylar commented:
@qingli411, thanks for fixing so many typos! I wish I were surprised. But I know that's just how it goes for me.

xylar commented:
I updated the tutorial following the fix in #144.  I also fixed a bunch more typos and replaced all tabs with spaces (even when there are tabs in the legacy COMPASS files).

xylar commented:
@matthewhoffman, do you have a sense of when you might have time to review this?  It's not a big rush but I also don't want it to get forgotten.

matthewhoffman commented:
I've had it sitting open in a tab for the past week.  I'll try to buckle down and look at it today or tomorrow.

matthewhoffman commented on docs/tutorials/dev_porting_legacy.rst:
```diff
@@ -0,0 +1,1295 @@
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The remainder of the ``init`` step will consist of defining the ``run()``
+method that does the refal work of the step.  To make it easier for users to
```
refal typo

xylar replied:
```suggestion
method that does the real work of the step.  To make it easier for users to
```

matthewhoffman commented on docs/tutorials/dev_porting_legacy.rst:
```diff
@@ -0,0 +1,1295 @@
+the initial condition (and therefore the rest of the test case).
+
+Since the config file has the same name (``default``) as the test case, it will
+be included automatically when the config file produced when test case gets
```
```suggestion
be included automatically when the config file is produced when the test case gets
```

matthewhoffman commented on docs/tutorials/dev_porting_legacy.rst:
```diff
@@ -0,0 +1,1295 @@
+Symlinks are created to a plotting script and the output from the ``forward``
+step, and then the plot script is run.
+
+An identical approach could be used in ``compass`` but it is not the perferred
```
```suggestion
An identical approach could be used in ``compass`` but it is not the preferred
```

matthewhoffman commented on docs/tutorials/dev_porting_legacy.rst:
```diff
@@ -0,0 +1,1295 @@
+
+For the user's guide, create a file
+``docs/users_guide/<core>/test_groups/<test_group>.rst``.  In that file, you
+should describe what the test group and its test cases do in a way that would
```
```suggestion
should describe what the test group and what its test cases do in a way that would
```

matthewhoffman reviewed (approved):
This is a very useful tutorial.  I think it will be helpful to developers creating a new test case, even if they haven't used legacy COMPASS.

xylar commented on docs/tutorials/dev_porting_legacy.rst:
```diff
@@ -0,0 +1,1295 @@
+  # this one will take a while the first time
+  ./conda/create_compass_env.py --conda $HOME/miniconda3
+
+If all goes well, you will have a file named ``test_compass_1.0.0*.sh``, where
```
```suggestion
If all goes well, you will have a file named ``load_dev_compass_1.0.0*.sh``, where
```

xylar commented on docs/tutorials/dev_porting_legacy.rst:
```diff
@@ -0,0 +1,1295 @@
+Okay, with those details as a reference point from legacy COMPASS, let's jump
+into developing the new test group in ``compass``.  Use any method you like
+for editing code.  If you haven't settled on a method and are working on your
+own laptop or desktop, you may want to try and integrated development
```
```suggestion
own laptop or desktop, you may want to try an integrated development
```

xylar commented on docs/tutorials/dev_porting_legacy.rst:
```diff
@@ -0,0 +1,1295 @@
+
+    class Gotm(TestGroup):
+        """
+        A test group for baroclinic channel test cases
```
```suggestion
        A test group for General Ocean Turbulence Model (GOTM) test cases
```

xylar commented:
@matthewhoffman, @qingli411 and @mark-petersen, thanks for the reviews!  I'll merge this as soon as the tests have finished.
